### ISSUE TITLE
Symbolic formula is incorrect when using multiplication nodes with mult_arity=4

### ISSUE THREAD

**Issue by chbardel** [KindXiaoming/pykan#404]
It seems the formula construction does not yet consider the different multarity cases correctly. My KAN is:
**width**=[5,[1,2],1]
**mult_arity**=4
**grid**=8
**k**=3

I started looking into the code of the MultKAN.**symbolic_formula**() function and compared it against the code in the MultKAN.**forward()**. Too me the handling of multiplication looks a lot more sophisticated in the forward pass and quite different from how its handled in the symbolic formula construction. I see a fixed stride length of two in the latter, which seems to suggest it does not handle the cases in which 4 nodes are multiplied with each other.

End result is that the formula, when applied to the inputs, gives a result different from the forward pass, although everything is set to fully symbolic mode.

If any further details are needed please let me know.

**Comment by KindXiaoming** [comment]
could you please show the KAN diagram (plot()) and the information that's been printed when you called auto_symbolic(). It's likely that some splines do not fit well with any symbolic function (indicated by low R2).

**Comment by chbardel** [comment]
Yes thats true. Some splines are set to zero even. But shouldnt that effect the formula and forward pass equally? I imagined it as simply setting one edge to zero, but as long as there are other edges the value at the summation node should still be nonzero. And indeed training is no problem.
But you are saying having symbolic edges being zero is handled differently in the formula vs. the forward pass?

**Here is the requested info**:
> fixing (0,0,0) with 1/x^0.5, r2=0.9989483952522278, c=2
fixing (0,0,1) with 0, r2=0.0, c=0
fixing (0,0,2) with 0, r2=0.0, c=0
fixing (0,0,3) with x, r2=0.9550677537918091, c=1
fixing (0,0,4) with 0, r2=0.0, c=0
fixing (0,0,5) with x, r2=0.9986269474029541, c=1
fixing (0,0,6) with 0, r2=0.0, c=0
fixing (0,0,7) with 0, r2=0.0, c=0
fixing (0,0,8) with x, r2=0.997905433177948, c=1
fixing (0,1,0) with 0, r2=0.0, c=0
fixing (0,1,1) with 0, r2=0.0, c=0
fixing (0,1,2) with 0, r2=0.0, c=0
fixing (0,1,3) with 0, r2=0.0, c=0
fixing (0,1,4) with 0, r2=0.0, c=0
fixing (0,1,5) with x^2, r2=0.9966174364089966, c=2
fixing (0,1,6) with 0, r2=0.0, c=0
fixing (0,1,7) with 0, r2=0.0, c=0
fixing (0,1,8) with x^2, r2=0.9992889761924744, c=2
fixing (0,2,0) with x, r2=0.9999688267707825, c=1
fixing (0,2,1) with x, r2=0.9523282647132874, c=1
fixing (0,2,2) with x, r2=0.9512811899185181, c=1
fixing (0,2,3) with x, r2=0.994516909122467, c=1
fixing (0,2,4) with x, r2=0.9782763123512268, c=1
fixing (0,2,5) with x, r2=0.9989272952079773, c=1
fixing (0,2,6) with x, r2=0.9984557628631592, c=1
fixing (0,2,7) with x, r2=0.9974117279052734, c=1
fixing (0,2,8) with x, r2=0.999871551990509, c=1
fixing (0,3,0) with x^2, r2=0.9976300001144409, c=2
fixing (0,3,1) with 0, r2=0.0, c=0
fixing (0,3,2) with 0, r2=0.0, c=0
fixing (0,3,3) with x, r2=0.9568336009979248, c=1
fixing (0,3,4) with 0, r2=0.0, c=0
fixing (0,3,5) with x, r2=0.9901322722434998, c=1
fixing (0,3,6) with x, r2=0.9910508990287781, c=1
fixing (0,3,7) with 0, r2=0.0, c=0
fixing (0,3,8) with x, r2=0.982476532459259, c=1
fixing (0,4,0) with x, r2=0.9999858736991882, c=1
fixing (0,4,1) with x, r2=0.9404085874557495, c=1
fixing (0,4,2) with x, r2=0.9530202150344849, c=1
fixing (0,4,3) with 1/x^2, r2=0.9995090365409851, c=2
fixing (0,4,4) with x, r2=0.9652431011199951, c=1
fixing (0,4,5) with x, r2=0.9998018145561218, c=1
fixing (0,4,6) with x, r2=0.9993450045585632, c=1
fixing (0,4,7) with sin, r2=0.9968628287315369, c=2
fixing (0,4,8) with x, r2=0.9998099207878113, c=1
fixing (1,0,0) with x, r2=0.9999396204948425, c=1
fixing (1,1,0) with 1/x^0.5, r2=0.9993799924850464, c=2
fixing (1,2,0) with x, r2=0.9662681221961975, c=1

![Screenshot (1)](https://github.com/user-attachments/assets/a1e98566-8587-4c98-bff9-c31b54b2d427)
